Content.Server/Theta/Impostor/Systems/ImpostorMagicRevolverSystem.cs

```diff
@@ -0,0 +1,163 @@
+using Content.Shared.Database;
+using Content.Shared.DoAfter;
+using Content.Shared.Examine;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Mind;
+using Content.Shared.Mind.Components;
+using Content.Shared.Mobs;
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Popups;
+using Content.Shared.Projectiles;
+using Content.Shared.Roles.Jobs;
+using Content.Shared.Theta.Impostor.Components;
+using Content.Shared.Weapons.Ranged.Components;
+using Content.Shared.Weapons.Ranged.Systems;
+using Content.Shared.Theta.Impostor.Events;
+using Robust.Shared.Map;
+using Robust.Shared.Network;
+
+namespace Content.Server.Theta.Impostor.Systems;
+
+public sealed class ImpostorMagicRevolverSystem : EntitySystem
+{
+    [Dependency] private PopupSystem _popupSys = default!;
+    [Dependency] private DoAfterSystem _doAfterSys = default!;
+    [Dependency] private SharedGunSystem _gunSys = default!;
+    [Dependency] private JobSystem _jobSys = default!;
+    [Dependency] private IBanManager _banMan = default!;
+    [Dependency] private MobStateSystem _mobSys = default!;
+    [Dependency] private MindSystem _mindSys = default!;
+
+    private const string MagicProjectileProtoId = "ImpostorMagicBullet";
+    private const string RegularProjectileProtoId = "BulletMagnum";
+    private const string CaptainJobId = "Captain";
+    private const string CommRelayProtId = "ImpostorCommRelay";
+    private const uint MagicBanDuration = 60*8; //minutes
+    
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ImpostorMagicBulletComponent, ExaminedEvent>(OnExamine);
+        SubscribeLocalEvent<ImpostorMagicBulletComponent, UseInHandEvent>(OnBulletUseInHand);
+        SubscribeLocalEvent<ImpostorMagicBulletComponent, ProjectileHitEvent>(OnBulletHit);
+        SubscribeLocalEvent<ImpostorMagicRevolverComponent, InteractUsingEvent>(OnRevolverUse, before: new[] { typeof(SharedGunSystem) });
+        SubscribeLocalEvent<ImpostorMagicRevolverComponent, ImpostorMagicBulletLoadedEvent>(OnBulletDoAfterFinish);
+        SubscribeLocalEvent<ImpostorMagicRevolverComponent, AttemptShootEvent>(OnRevolverShoot);
+    }
+
+    private void OnExamine(EntityUid uid, ImpostorMagicBulletComponent bullet, ExaminedEvent args)
+    {
+        if (HasComp<ProjectileComponent>(uid))
+            return;
+        
+        if(bullet.Marked)
+            args.PushText(Loc.GetString("impostor-magicbullet-marked"));
+    }
+    
+    private void OnBulletUseInHand(EntityUid uid, ImpostorMagicBulletComponent bullet, UseInHandEvent args)
+    {
+        if (bullet.Marked)
+        {
+            _popupSys.PopupEntity(Loc.GetString("impostor-magicbullet-unmark"), uid, args.User);
+        }
+        else
+        {
+            _popupSys.PopupEntity(Loc.GetString("impostor-magicbullet-mark"), uid, args.User);
+        }
+        bullet.Marked = !bullet.Marked;
```

Comment:
```suggestion
	        bullet.Marked = !bullet.Marked;
        if (bullet.Marked)
        {
            _popupSys.PopupEntity(Loc.GetString("impostor-magicbullet-mark"), uid, args.User);
        }
        else
        {
            _popupSys.PopupEntity(Loc.GetString("impostor-magicbullet-unmark"), uid, args.User);
        }
```
Тоже чисто из-за читаемости, чтобы было иф маркет { маркет }, а не отрицание